Topics/Development_Process/Azure_webapp_deployment_with_nodejs.md

```diff
@@ -1,10 +1,191 @@
+    - Once the command successfully executes, visit the URL in the response object you got. If you see your web app (note that it may take a minute or two), congratulations, you've successfully deployed your web app on Azure!
+
+## CI/CD with Github Actions
+
+You can use Github Actions with Azure to automate workflows during your software development. Given the different hosting and deployment options mentioned above, there is a different way to use Github Actions to work with whichever hosting option you've chosen.
+
+Below, we'll continue with our previous example of a Node.js Web App using Azure App Service, but you can find examples for different hosting options [here](https://learn.microsoft.com/en-us/azure/developer/github/deploy-to-azure).
+
+Note that before continuing with implementing CI/CD with Github Actions, you should have a working [Azure App Service](#azure-app-service) app.
+
+### Using Github Actions to Deploy to Azure App Service
+
+The workflow can be set up in 1 of 2 ways:  using the Deployment Center to generate the workflow file for your Azure App service and tech stack in a Github repository, or manually.
+
+If you deployed your web app using the Azure portal, congratulations, the workflow file was set up automatically for you! You're all done! If you deployed your web app using the Azure CLI, follow along, there's a couple more steps!
+
+I found the official documentation (found at the link below) to be a clear and quick tutorial for how to set up Github Actions for your Azure App Service web app, which guides you through whichever workflow setup method you've chosen, with instructions for the various supported programming languages you may be using.
+
+[Using Github Actions with Azure App Service](https://learn.microsoft.com/en-us/azure/app-service/deploy-github-actions?tabs=applevel)
+
+Some issues that I've seen occur in this process include:
+
+- The workflow script not being able to find a file, in which case you may want to modify your file structure or the workflow file itself to properly reference the location.
+- Npm script errors if you haven't set up a test script. You can fix this by removing the calls to these scripts (ie. remove `npm run test` and `npm run build` from the workflow file if you don't have these)
+
+Once you've completed the steps in the documentation and see that the Github Actions workflow ran successfully, congratulations, you've deployed a web app on Azure integrated with Github Actions for CI/CD!
+
+## Cost Management with Azure
+
+There are multiple scary stories of people learning to use cloud service providers, like Google Cloud Platform or AWS, and as they set up different services, when the bill comes around, they're faced with a staggeringly large bill. 
+
+It can be scary to learn something new when such consequences are possible, but to avoid that, Azure provides a handful of tools to use to ensure that you are in control of what you spend.
```

Comment:
What can you spend money on on Azure? Perhaps tell us what this deployment can lead to as an example

Reply:
Given that it's not the main focus of this article, I won't be going into immense detail about everything that a user could be paying for on Azure. I added this section of the article as it's something that I've always been worried about when using a cloud service provider, and I wanted to make the reader aware to inform them of preventative measures to take. 
However, I agree that a bit more of an explanation about what mistakes can lead to larger bills could help put the user at ease, so I've added a brief explanation about how these charges can come from inexperience with Azure and its provided resources, and about how our choice of resources were done to minimize the risk of a large bill building up.